This submission can get you up to 5 extra points on your midterm. It is due November 6th at 6:30 PM sharp - nothing submitted after that will be accepted. Please submit it to my email with the subject 'Extra Credit Assignment \#1'

In [1]:
import scipy.stats as stats
import numpy as np

1. Say that we flip 10 coins 100 times. What is the expected number of times we will see 0 - 10 heads in our simulation? (i.e. we'll see 5 heads 100 times, or 0 heads 20 times and 10 heads 80 times)

In [2]:
expected_counts = []
expected_values = []
for i in range(11):
    expected_values.append(i)
    expected_counts.append(stats.binom(p=0.5, n=10).pmf(i) * 100)
print('Expected Results of One-Hundred Dice Rolls')
np.transpose((expected_values, expected_counts))

Expected Results of One-Hundred Dice Rolls


array([[ 0.        ,  0.09765625],
       [ 1.        ,  0.9765625 ],
       [ 2.        ,  4.39453125],
       [ 3.        , 11.71875   ],
       [ 4.        , 20.5078125 ],
       [ 5.        , 24.609375  ],
       [ 6.        , 20.5078125 ],
       [ 7.        , 11.71875   ],
       [ 8.        ,  4.39453125],
       [ 9.        ,  0.9765625 ],
       [10.        ,  0.09765625]])

2. Now say I have the below observation. At a signifiance level of 0.05, can I reject the null hypothesis that the observation comes from the expected distribution?

In [3]:
observed_counts = [0,1,6,19,16,32,15,4,3,4,0]

In [4]:
stats.chisquare(observed_counts, expected_counts)

Power_divergenceResult(statistic=24.883301587301563, pvalue=0.005571147877628013)

You can reject the null hypothesis since this result has a p-value of less than 0.05.

3. At what value can I reject the null hypothesis?

In [5]:
stats.chi2(df=10).ppf(0.95)

18.307038053275146

You can reject the null hypothesis if the Chi-square statistic is anything greater than 18.3.